docs/source/nlp/nemo_megatron/flash_attention.rst

```diff
@@ -0,0 +1,19 @@
+Flash attention
+---------------
+`FlashAttention <https://arxiv.org/pdf/2205.14135.pdf>`_ is a method designed to enhance the efficiency of Transformer models, which are widely utilized in applications such as natural language processing. Traditional Transformers are slow and consume a lot of memory, especially with long sequences, due to the quadratic time and memory complexity of self-attention. FlashAttention, an IO-aware exact attention algorithm that leverages tiling to minimize the number of memory reads/writes between the GPU's high bandwidth memory (HBM) and on-chip SRAM. This approach is designed to be more efficient in terms of IO complexity compared to standard attention mechanisms.
+
+GPT
+^^^
+To enable Flash Attention while Megatron GPT model training or fine-tuning, modify the following configuration: 
+
+.. code::
+
+   model.use_flash_attention=True
+
+T5
+^^
+To enable Flash Attention while Megatron T5 model training, modify the following configuration: 
+
+.. code::
+
+   model.use_flash_attention=True
```

Comment:
don't we need to set it both in encoder and decoder separately?
@ssh-meister @hsiehjackson

Reply:
Thank you! Fixed!